### PR TITLE
Ftreguer restful api

### PR DESCRIPTION
**Pull request by ftreguer**
Add restful API

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,8 +30,79 @@ Or launch backend with automatic dev reload:
...
 
+
+Launch tests:
+```
+yarn test
+```
+
 The API will be available at `http://localhost:8080/`
 
+### API Routes
```
Comment (dlecan) [inline README.md:41]: :heart: I like the way you have documented the API

You are ready for https://swagger.io/specification/ Swagger3-like specifications :laughing:

File: package.json [hunk 2]
```diff
@@ -1,27 +1,64 @@
...
-  "author": "sonarplanet",
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:sonarplanet/sonarplanet-services.git"
+  },
+  "bugs": {
+    "url": "https://github.com/sonarplanet/sonarplanet-services/issues"
+  },
+  "homepage": "https://github.com/sonarplanet/sonarplanet-services/README.md",
+  "author": "TREGUER Fabien <fabien.treguer@spacelephant.org> http://spacelephant.org#fabien",
```
Comment (dlecan) [inline package.json]: :heart:  Link to your profile

File: src/routers/router.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                  console.error("Error adding public address subscription on account. " + errUPT.message)
+                  res.status(500).send()
+                } else {
+                  if (account.webPushNotification) {
+                    database.webPushNotificationCollection.findOne({ _id: account.webPushNotification }, (err: Error, wpn: any) => {
+                      if (err) {
+                        console.error("Error fetching webpush notification parameters for account " + account.uniqueId + ' ' + err.message)
+                        res.status(500).send()
+                      } else {
+                        TrackerUtils.watchWebPush(networkId, publicAddress, getWebPushSubstriptionObject(wpn))
```
Comment (dlecan) [inline src/routers/router.ts]: Callback hell! http://callbackhell.com/
Let's refactor that in the next Sprint

File: src/routers/router.ts [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+  let accountId = req.params.accountId
+  database.accountCollection.findOne({ uniqueId: accountId }, (err: Error, account: any) => {
+    if (err) {
+      console.error("Error occured fetching account " + accountId + ' ' + err.message)
+      res.status(500).send()
+    }
+    if (account) {
+      let webPushNotif
+      try {
+        webPushNotif = {
```
Comment (dlecan) [inline src/routers/router.ts:153]: Why not just `webPushNotif = req.body.subscription`?

File: src/routers/router.ts [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+  })
+})
+
+// Get all networks
+//networkRouter.get('/', (req: any, res: any) => {})
+
+// Get all enabled medias for an account
+//enabledNotificationMediaRouter.get('/', (req: any, res: any) => {})
+
+let getWebPushSubstriptionObject = (webPushNotificationParameters: any) => {
```
Comment (dlecan) [inline src/routers/router.ts]: Misspelling

File: src/trackerUtils.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import * as trackr from 'trackr-lib'
+import * as webpush from 'web-push'
+
+// VAPID keys should only be generated only once.
+const vapidKeys = webpush.generateVAPIDKeys();
+const options = {
+  vapidDetails: {
+    subject: 'mailto:support@sonarplanet.io',
```
Comment (dlecan) [inline src/trackerUtils.ts]: support@... Excellent! It works since ... today :smile:

**Review by dlecan — CHANGES_REQUESTED** [review]
Good work! :heart_eyes: 

Few things to update.

:warning:  endless test suite will block your PR after rebase on #3

File: src/routers/router.ts (hunk 5 above)
Comment (ftreguer) [reply src/routers/router.ts]: mappWebPushSubscriptionObject ?

Comment (dlecan) [reply src/routers/router.ts]: `getWebPushSubstriptionObject` => `getWebPushSubscriptionObject`

Comment (ftreguer) [reply src/routers/router.ts]: :D You don't like Substription ? :smile:

File: src/routers/router.ts (hunk 4 above)
Comment (ftreguer) [reply src/routers/router.ts:153]: Key level is not needed. I will change the body client-side to remove key level and directly get subscription server-side

**Review by dlecan — APPROVED** [review]

**Pull request merged**